pytype/typegraph/cfg.cc

```diff
@@ -499,51 +499,52 @@ PyDoc_STRVAR(
     "stay in memory.");
 
 PyTypeObject PyProgram = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0) tp_name : "Program",
-  tp_basicsize : sizeof(PyProgramObj),
-  tp_itemsize : 0,
-  tp_dealloc : ProgramDealloc,
+  PyVarObject_HEAD_INIT(&PyType_Type, 0) 
```

Comment:
msvc added support for designated initializers in 2019; what would the impact of not making this change be? (i'm fine  with leaving it in if a significant number of users would be impacted, or if it is onerous to upgrade msvc for people who have an older version, but it makes the code more fragile)

Reply:
Mixing old-style syntax and designated initializer together is [not supported](https://en.cppreference.com/w/cpp/language/aggregate_initialization#:~:text=Note%3A%20out%2Dof%2Dorder%20designated%20initialization%2C%20nested%20designated%20initialization%2C%20mixing%20of%20designated%20initializers%20and%20regular%20initializers%2C%20and%20designated%20initialization%20of%20arrays%20are%20all%20supported%20in%20the%20C%20programming%20language%2C%20but%20are%20not%20allowed%20in%20C%2B%2B.) by neither c++20 standard nor msvc. I guess there are some parameters written in old-style syntax in `PyVarObject_HEAD_INIT`.   Original code cause compile-time errors in VS19/22 even if language standard is set to c++20 or latest.

Reply:
Sure. It will be excellent if designated initializers can be kept. I have restored this file.